contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/BaseTestHiveImpersonation.java

```diff
@@ -97,11 +99,29 @@ protected static void prepHiveConfAndData() throws Exception {
   }
 
   protected static void startHiveMetaStore() throws Exception {
-    final int port = MetaStoreUtils.findFreePort();
+    Class<?> metaStoreUtilsClass;
+    Class<?> hadoopThriftAuthBridgeClass;
+    Class<?> confClass;
+    Object hadoopThriftAuthBridge;
+    // TODO: remove reflection stuff when all supported profiles will be switched to Hive 3+ version
```

Comment:
It would be good to add Jira ticket to the comment.

Reply:
Currently, we support only a single non-default profile, but it hasn't released Hive 3+ versions, so I think that it would be better to postpone the creation of the Jira to the time when it is released (if it will be released), so we will know the exact version.

But I have created another Jira in Hive - HIVE-23088 to fix and remove workaround with logger class.